### PR TITLE
Stop using deprecated aliases of builtin types

### PR DESCRIPTION
**Pull request by akuhnregnier**
### Description
This PR removes deprecated NumPy aliases of builtin types, which is required to avoid warnings starting with NumPy 1.20.0.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: lib/iris/tests/test_cell.py [hunk 1]
```diff
@@ -130,15 +130,15 @@ class Terry:
         self.assertEqual(self.d.__ne__(Terry()), NotImplemented)
 
     def test_numpy_int_equality(self):
-        dtypes = (np.int, np.int16, np.int32, np.int64)
+        dtypes = (int, np.int16, np.int32, np.int64)
```
Comment (akuhnregnier) [inline lib/iris/tests/test_cell.py:133]: Given that this method name contains 'numpy' I'm not sure if a bare `int` should remain here now.

File: lib/iris/tests/test_cell.py [hunk 2]
```diff
@@ -130,15 +130,15 @@ class Terry:
...
         for dtype in dtypes:
             val = dtype(3)
             cell = iris.coords.Cell(val, None)
             self.assertEqual(cell, val)
 
     def test_numpy_float_equality(self):
         dtypes = (
-            np.float,
+            float,
```
Comment (akuhnregnier) [inline lib/iris/tests/test_cell.py:141]: Given that this method name contains 'numpy' I'm not sure if a bare `float` should remain here now.

File: lib/iris/tests/test_cell.py (hunk 1 above)
Comment (bjlittle) [reply lib/iris/tests/test_cell.py:133]:
```suggestion
        dtypes = (np.int_, np.int16, np.int32, np.int64)
```

Comment (bjlittle) [reply lib/iris/tests/test_cell.py:133]: @akuhnregnier Great spot and attention to detail :eyes: 

I think honouring the intent of the test, we could opt for `np.int_` :+1:

File: lib/iris/tests/test_cell.py (hunk 2 above)
Comment (bjlittle) [reply lib/iris/tests/test_cell.py:141]:
```suggestion
            np.float_,
```

Comment (bjlittle) [reply lib/iris/tests/test_cell.py:141]: @akuhnregnier Using `np.float_` might be a reasonable substitute here :+1:

**Review by bjlittle — CHANGES_REQUESTED** [review]
Hi @akuhnregnier,

Thanks for your contribution! Much appreciated indeed :+1: 

I made a couple of suggestions. So if you're happy with those suggested changes, then this LGTM.

Otherwise, it would be awesome to recognise your contribution by making an entry our documentations `whatsnew`.

Simply make an entry in the `docs/src/whatsnew/latest.rst`, under the **💼 Internal** section, something along the lines of the following:
```
#. `@akuhnregnier`_ replaced `deprecated numpy 1.20 aliases for builtin types`_.
   (:pull:`3997:`)
``` 
Then add the `rst` directives for your github name and the numpy docs references into the appropriate sections at the end of the `latest.rst` e.g.,
```
.. _@akuhnregnier: https://github.com/akuhnregnier
```
and
```
.. _deprecated numpy 1.20 aliases for builtin types: https://numpy.org/doc/1.20/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated
```
Awesome, thank you :tada: :100: :beers:

**Comment by akuhnregnier** [comment]
Hi @bjlittle,

Thank you for taking a look - I'm glad this is helpful!

If there is anything else I can do to help get this merged, please let me know.

**Comment by bjlittle** [comment]
@akuhnregnier Awesome! 👍

Okay, some changes just got merged into `master` that will fix the failing `linkcheck` tests on this pull-request, see #4003.

So simply rebase to pull those changes into your local development branch e.g.,
```bash
> git checkout deprecated-builtin-types
> git fetch upstream
> git rebase upstream/master
> git push origin +HEAD
```
Obviously, if there are any conflicts you'll need to resolve them, add and commit the changes before pushing to your remote tracking branch.

If anything is in conflict, I suspect it might be the `docs/src/whatsnew/latest.rst`... But hopefully it's not a problem or straight forward for you to figure out

**Review by bjlittle — APPROVED** [review]

**Comment by bjlittle** [comment]
@akuhnregnier Congrats! :partying_face: :tada: :beers: 

And welcome the `iris` family!

We really do appreciate your first contribution to `iris`... and long may they continue :wink: 

... which reminds me, I'll take a peek at #3994 when I get an opportunity :+1:

**Comment by akuhnregnier** [comment]
Thanks @bjlittle - I've been using `iris` for quite a while so I'm very happy to be a (small) part of it now! :tada:

**Pull request merged**